src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/SearchObject.cs

```diff
@@ -233,6 +235,51 @@ private void CacheBackgroundData(IPackageSearchMetadata packageSearchMetadata, b
             {
                 memoryCacheItem.UpdateSearchMetadata(packageSearchMetadata);
             }
+
+            Assumes.NotNull(packageSearchMetadata);
```

Comment:
Should do this check sooner since it is used higher up.

Reply:
moving to after the first if block... make sense to be that high?